src/app/password-bars/password-bars.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, OnInit, Input, SimpleChange, Output, EventEmitter } from '@angular/core';
+import { reduce } from 'rxjs';
+
+
+@Component({
+  selector: 'app-password-bars',
+  templateUrl: './password-bars.component.html',
+  styleUrls: ['./password-bars.component.scss']
+})
+export class PasswordBarsComponent {
+  @Input() public passwordToCheck: string;
+  @Output() passwordStrength = new EventEmitter<boolean>();
+  private colors = ['red', 'yellow', 'green'];
+  bar1: string;
+  bar2: string;
+  bar3: string;
+
+  checkStrength(password) {
+    let strength = 0;
+    strength = password.length >= 8 ? strength += 10 : 0;
+
+    const letters = /[a-zA-Z]+/.test(password);
+    const numbers = /[0-9]+/.test(password);
+    const regex = /[$-/:-?{-~!"^_@`\[\]]/g;
+    const symbols = regex.test(password);
+
+    const flags = [letters, numbers, symbols];
+
+    let passedMatches = 0;
+    for (const flag of flags) {
+      passedMatches += flag === true ? 1 : 0;
```

Comment:
```suggestion
      passedMatches += flag ? 1 : 0;
```
imho it's better to use `if` and `passedMatches++`